src/FixedDataTableBufferedRows.js

```diff
@@ -77,119 +77,96 @@ class FixedDataTableBufferedRows extends React.Component {
-        }
-        continue;
+        rowIndex = this._staticRowArray[i] && this._staticRowArray[i].props.index;
       }
 
-      const currentRowHeight = this.props.rowSettings.rowHeightGetter(rowIndex);
-      const currentSubRowHeight = this.props.rowSettings.subRowHeightGetter(rowIndex);
-      const rowOffsetTop = baseOffsetTop + props.rowOffsets[rowIndex];
-      const rowKey = props.rowKeyGetter ? props.rowKeyGetter(rowIndex) : i;
-      const hasBottomBorder = (rowIndex === props.rowSettings.rowsCount - 1) &&
-        props.showLastRowBorder;
-      const visible = this.isRowInsideViewport(rowIndex);
-
-      this._staticRowArray[i] =
-        <FixedDataTableRow
-          key={rowKey}
-          isScrolling={props.isScrolling}
-          index={rowIndex}
-          width={props.width}
-          height={currentRowHeight}
-          subRowHeight={currentSubRowHeight}
-          rowExpanded={props.rowExpanded}
-          scrollLeft={Math.round(props.scrollLeft)}
-          offsetTop={Math.round(rowOffsetTop)}
-          visible={visible}
-          fixedColumns={props.fixedColumns}
-          fixedRightColumns={props.fixedRightColumns}
-          scrollableColumns={props.scrollableColumns}
-          onClick={props.onRowClick}
-          onContextMenu={props.onRowContextMenu}
-          onDoubleClick={props.onRowDoubleClick}
-          onMouseDown={props.onRowMouseDown}
-          onMouseUp={props.onRowMouseUp}
-          onMouseEnter={props.onRowMouseEnter}
-          onMouseLeave={props.onRowMouseLeave}
-          onTouchStart={props.onRowTouchStart}
-          onTouchEnd={props.onRowTouchEnd}
-          onTouchMove={props.onRowTouchMove}
-          showScrollbarY={props.showScrollbarY}
-          className={joinClasses(
-            rowClassNameGetter(rowIndex),
-            cx('public/fixedDataTable/bodyRow'),
-            cx({
-              'fixedDataTableLayout/hasBottomBorder': hasBottomBorder,
-              'public/fixedDataTable/hasBottomBorder': hasBottomBorder,
-            })
-          )}
-        />;
+      this._staticRowArray[i] = this.renderRow({
+        rowIndex,
+        key: i,
+        baseOffsetTop,
+      });
     }
 
-    return <div>{this._staticRowArray}</div>;
+    return <div> {this._staticRowArray} </div>;
   }
 
-
   /**
-   * Returns a stub row which won't be visible to the user.
-   * This allows us to still render a row and React won't unmount it.
-   *
+   * @param {number} rowIndex
    * @param {number} key
-   * @param {boolean} fake
-   * @param {number} index
+   * @param {number} baseOffsetTop
    * @return {!Object}
    */
-  getStubRow(key, fake, index) /*object*/ {
+  renderRow({ rowIndex, key, baseOffsetTop }) /*object*/ {
     const props = this.props;
+    const rowClassNameGetter = props.rowClassNameGetter || emptyFunction;
+    const fake = rowIndex === undefined;
+    let rowProps = {};
+
+    // if row exists, then calculate row specific props
+    if (!fake) {
```

Comment:
this was were we had trouble (actual issue)

previously rows outside the viewport were seen as stub/fake, and hence given bad props